exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -492,7 +538,19 @@ private void writeDataAllText(MapWriter map, FieldSelection selection,
       }
     }
     map.end();
+  }
 
+  /**
+   * Puts copy of field path list to fieldPathWriter map.
+   * @param fieldName
+   */
+  private void putFieldPath(String fieldName, MapWriter map) {
```

Comment:
Do you have the performance data that you had collected earlier with different percentage of NULLs, after this change ?  I still feel there could be some further optimization needed.